### PR TITLE
Add new details about HA - (with new release notes only)

### LINKED ISSUE #2972: Details to HA Deployments on Hosted
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://docs.astronomer.io/astro/deployment-settings#enable-high-availability

### The Issue or Context

Add details to describe how HA is achieved for the `pgbouncer` and `scheduler` components with HA enabled. We set node anti affinity to `required` guaranteeing that a replica of each runs on separate nodes. Additionally, we are also introducing `preferred` zone anti affinity for best effort scheduling of the replicas on separate availability zones as well.

https://github.com/astronomer/astro/pull/16673#event-10700577225

### Fix or Proposed Change

_No response_

### Additional Notes

_No response_

### Required Reviewers

chomipi88, pritt20

### PR DESCRIPTION
**Pull request by manmeetkaur**
Resolves #2972

### CODE REVIEW

File: astro/deployment-settings.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ To configure the scheduler on an [Astro Hybrid](hybrid-overview.md) Deployment:
 
 ### Enable high availability
 
-By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of its scheduler across different nodes. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Note that PGBouncer Pods are guaranteed to run across multiple availability zones, but scheduler Pods are not. 
+By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of Airflow scheduler across different nodes and availability zones. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node or availability zone.
```
Comment (lzdanski) [inline astro/deployment-settings.md:225]:
```suggestion
By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of the Airflow scheduler across different nodes and availability zones. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node or availability zone. 
```

**Review by lzdanski — APPROVED** [review]
Tiniest suggestion!

File: astro/deployment-settings.md [hunk 2]
```diff
@@ -222,7 +222,7 @@ To configure the scheduler on an [Astro Hybrid](hybrid-overview.md) Deployment:
 
 ### Enable high availability
 
-By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of its scheduler across different nodes. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Note that PGBouncer Pods are guaranteed to run across multiple availability zones, but scheduler Pods are not. 
+By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of the Airflow scheduler across different nodes and availability zones. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node or availability zone.
```
Comment (chomipi88) [inline astro/deployment-settings.md:225]: different nodes is **guaranteed** aka required for k8s anti affinity rules
different availability zones is **best effort** aka preferred for k8a anti affinity rules

based on our testing, the scheduler and pgbouncer pods will most likely land on different AZs but we cannot guarantee it.

Comment (lzdanski) [inline astro/deployment-settings.md:225]:
```suggestion
By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of the Airflow scheduler across different nodes. However, Astro makes a best effort to use different availability zones for your PGBouncer and scheduler, which means it is possible but unlikely that they are both located in the same availability zone. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node or availability zone. 
```

Comment (chomipi88) [reply astro/deployment-settings.md:225]: LGTM

**Review by chomipi88 — APPROVED** [review]

File: astro/deployment-settings.md (hunk 2 above)
Comment (chomipi88) [reply astro/deployment-settings.md:225]: Why don't we capitalize Scheduler?

Comment (lzdanski) [reply astro/deployment-settings.md:225]: Whoops good catch! Fixing

**Pull request merged**